## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 466959 -p superlu_dist.tests.mpich -p superlu_dist`
Commit: [`359738926978e06bfab34aec92f6e56fde740f95`](https://github.com/NixOS/nixpkgs/commit/359738926978e06bfab34aec92f6e56fde740f95) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/359738926978e06bfab34aec92f6e56fde740f95..pull/466959/head))
Merge: [`bff6c375032481a4a3b699b7435d7ab7fe9989b2`](https://github.com/NixOS/nixpkgs/commit/bff6c375032481a4a3b699b7435d7ab7fe9989b2)

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/19838906236


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>superlu_dist</li>
    <li>superlu_dist.tests.mpich</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>superlu_dist</li>
    <li>superlu_dist.tests.mpich</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>superlu_dist.tests.mpich</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>superlu_dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>superlu_dist.tests.mpich</summary>
<pre>cmake: enabled parallel building
cmake: enabled parallel installing
configurePhase completed in 2 minutes 39 seconds
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/lbkni2ik42yc5hic017z8r6h5gn1gy04-bash-5.3p3/bin/bash
[  1%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_ienv.c.o
[  1%] Generating superlu_dist_config.fh
[  1%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/etree.c.o
[  1%] Built target config_f
[  2%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_colorder.c.o
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:79: SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_ienv.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:93: SRC/CMakeFiles/superlu_dist.dir/prec-independent/etree.c.o] Error 1
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:107: SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_colorder.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1101: SRC/CMakeFiles/superlu_dist.dir/all] Error 2
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>superlu_dist.tests.mpich</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>superlu_dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>superlu_dist.tests.mpich</summary>
<pre>-- Build files have been written to: /nix/build/nix-38101-2142777213/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/p0k9r5h8qs7220xdbdihhfgzwjcly70x-bash-5.3p3/bin/bash
[  0%] Generating superlu_dist_config.fh
[  0%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/etree.c.o
[  1%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_ienv.c.o
[  1%] Built target config_f
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
[  2%] Building C object SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_colorder.c.o
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:93: SRC/CMakeFiles/superlu_dist.dir/prec-independent/etree.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:79: SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_ienv.c.o] Error 1
clang: error: unknown argument: &#x27;-fallow-argument-mismatch&#x27;
make[2]: *** [SRC/CMakeFiles/superlu_dist.dir/build.make:107: SRC/CMakeFiles/superlu_dist.dir/prec-independent/sp_colorder.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1101: SRC/CMakeFiles/superlu_dist.dir/all] Error 2
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/su/superlu_dist/package.nix
+++ b/pkgs/by-name/su/superlu_dist/package.nix
@@ -18,6 +18,7 @@
   isILP64 ? false,
 
   # passthru.tests
+  mpich,
   superlu_dist,
 }:
 
@@ -35,6 +36,13 @@ stdenv.mkDerivation (finalAttrs: {
     postFetch = "rm $out/SRC/prec-independent/mc64ad_dist.c";
     hash = "sha256-i/Gg+9oMNNRlviwXUSRkWNaLRZLPWZRtA1fGYqh2X0k=";
   };
+
+  # --oversubscribe unrecognized by mpich
+  # see https://github.com/xiaoyeli/superlu_dist/issues/208
+  postPatch = ''
+    substituteInPlace TEST/CMakeLists.txt \
+      --replace-fail "-oversubscribe" ""
+  '';
 
   patches = [
     ./mc64ad_dist-stub.patch
@@ -89,6 +97,11 @@ stdenv.mkDerivation (finalAttrs: {
     inherit isILP64;
     tests = {
       ilp64 = superlu_dist.override { isILP64 = true; };
+    }
+    // lib.optionalAttrs stdenv.hostPlatform.isLinux {
+      mpich = superlu_dist.override {
+        mpi = mpich;
+      };
     };
   };
 

